apps/music/js/ui/views/tiles_view.js

```diff
@@ -42,44 +38,50 @@ var TilesView = {
   },
 
   clean: function tv_clean() {
-    // Cancel a pending enumeration before start a new one
-    if (this.handle) {
-      musicdb.cancelEnumeration(this.handle);
-    }
-
     this.dataSource = [];
     this.index = 0;
     this.anchor.innerHTML = '';
-    this.view.scrollTop = 0;
-    this.hideSearch();
+
+    // Don't reset the scroll position here, since we want to persist it
+    // every time the view is updated!
   },
 
-  hideSearch: function tv_hideSearch() {
+  hideSearch: function tv_hideSearch(behavior) {
     this.searchInput.value = '';
-    // XXX: we probably want to animate this...
-    if (this.view.scrollTop < this.searchBox.offsetHeight) {
-      this.view.scrollTop = this.searchBox.offsetHeight;
+    if (this.view.scrollTop < this.anchor.offsetTop) {
+      this.anchor.scrollIntoView({behavior: behavior || 'smooth'});
     }
   },
 
-  update: function tv_update(result) {
-    // if no songs in dataSource
-    // disable the TabBar to prevent users switch to other page
-    TabBar.setDisabled(!this.dataSource.length);
+  activate: function tv_activate(callback) {
+    // Enumerate existing song entries in the database. List them all, and
+    // sort them in ascending order by album. Use enumerateAll() here so that
+    // we get all the results we want and then pass them synchronously to the
+    // _addSong() function. If we did it asynchronously, then we'd get one
+    // redraw for every song.
 
-    if (result === null) {
-      App.showCorrectOverlay();
-      // Display the TilesView after when finished updating the UI
-      document.getElementById('views-tiles').classList.remove('hidden');
-      // After the hidden class is removed, hideSearch can be effected
-      // because the computed styles are applied to the search elements
-      // And ux wants the search bar to retain its position for about
-      // a half second, but half second seems to short for notifying users
-      // so we use one second instead of a half second
-      window.setTimeout(this.hideSearch.bind(this), 1000);
-      return;
+    // Cancel a pending enumeration before starting a new one.
+    if (this.handle) {
+      musicdb.cancelEnumeration(this.handle);
     }
 
+    this.handle = musicdb.enumerateAll('metadata.album', null, 'nextunique',
+                                       function(songs) {
+      TilesView.clean();
+      songs.forEach(function(song) {
+        TilesView._addSong(song);
+      });
+
+      // Display the TilesView once the UI has been populated.
+      document.getElementById('views-tiles').classList.remove('hidden');
+
+      if (callback) {
+        callback(songs);
+      }
+    });
+  },
+
+  _addSong: function tv_addSong(result) {
```

Comment:
Currently each tile represents one album, though a result does means a song record from the MediaDB, but do you think to use `_addAlbum` is more suitable? or maybe `_addItem` because if we change the design of the TilesView in the future, we don't need to change the function name and still make sense.